packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -605,6 +609,7 @@ export class GarbageCollector implements IGarbageCollector {
 
         // Whether we are running in test mode. In this mode, unreferenced nodes are immediately deleted.
         this.testMode = this.mc.config.getBoolean(gcTestModeKey) ?? this.gcOptions.runGCInTestMode === true;
+        this.tombstoneMode = this.gcOptions.tombstoneMode === true;
```

Comment:
You should check for the feature flag here instead of an option.
Also, consider passing in a parameter to deleteUnusedRoutes that tells whether to delete or tombstone. That way you don't have to make any changes in the container runtime and DataStores can handle whether to delete or tombstone in the "deleteUnusedRoutes" method.

Reply:
Updated to use a feature flag